src/java/main/org/apache/zookeeper/client/ZooKeeperSaslClient.java

```diff
@@ -590,14 +590,13 @@ public boolean clientTunneledAuthenticationInProgress() {
                 // authentication is either in progress, successful, or failed.
 
                 // 1. Authentication hasn't finished yet: we must wait for it to do so.
-                if ((isComplete() == false) &&
-                    (isFailed() == false)) {
+                if ((!isComplete()) && (!isFailed())) {
                     return true;
                 }
 
                 // 2. SASL authentication has succeeded or failed..
                 if (isComplete() || isFailed()) {
```

Comment:
Hi, @shralex. Yep, if we get the value of `!isComplete()` is `true`, then we already know the authentication has not yet been completed, so it should return `true` for the `clientTunneledAuthenticationInProgress` method directly.